util/util.go

```diff
@@ -0,0 +1,36 @@
+package util
+
+import (
+	"fmt"
+	"strings"
+)
+
+// SplitCmd split full command to namespace, command, version.
+// ex(ns/cmd/1.0.1 => ns cmd 1.0.1)
+func SplitCmd(cmd string) (namespace, command, version string, err error) {
+	splitNamespce := strings.Split(cmd, "/")
+	if len(splitNamespce) < 2 {
+		err = fmt.Errorf("Command format is not valid")
+		return
+	}
+	splitCmdAndVer := strings.Split(splitNamespce[1], "@")
+	if len(splitCmdAndVer) < 2 {
+		err = fmt.Errorf("Command format is not valid")
+		return
+	}
+	namespace = splitNamespce[0]
+	command = splitCmdAndVer[0]
+	version = splitCmdAndVer[1]
+	return
+}
+
+// SplitCmdWithSearch search full command. If there is valid full command, return splited full command name.
+func SplitCmdWithSearch(cmds []string) (namespace, command, version string, itr int, err error) {
+	for i, val := range cmds {
+		ns, cmd, ver, err := SplitCmd(val)
+		if err == nil {
+			return ns, cmd, ver, i, err
```

Comment:
I think these should be validated with splitting. For example, `ns/cmd@+*%"` is accepted now.

Reply:
You are right. Namespace and command has no information about validation but version have some rules. I will add validation of version.

Reply:
I set regular expression for these. 
Namespace and command  are allowed to use `[a-zA-Z0-9-_]` characters.
Version is also checked by regular expression. I check expression the [command design(Details.Versioning)](https://github.com/screwdriver-cd/screwdriver/blob/master/design/commands.md) said.